framework_tvb/tvb/core/services/project_service.py

```diff
@@ -295,16 +298,23 @@ def remove_project(self, project_id):
 
             self.logger.debug("Deleting project: id=" + str(project_id) + ' name=' + project2delete.name)
             project_datatypes = dao.get_datatypes_in_project(project_id)
+
+            # Delete datatypes one by one in the reversed order of their creation date
             project_datatypes.sort(key=lambda dt: dt.create_date, reverse=True)
             for one_data in project_datatypes:
                 self.remove_datatype(project_id, one_data.gid, True)
 
-            links = dao.get_links_for_project(project_id)
-            for one_link in links:
-                dao.remove_entity(Links, one_link.id)
-            project_bursts = dao.get_bursts_for_project(project_id)
-            for burst in project_bursts:
-                dao.remove_entity(burst.__class__, burst.id)
+            # Delete any links for datatypes that point to the project they are in because they are redundant.
+            # This can happen when a datatype is linked with more datatypes (e.g. a Connectivity linked to
+            # a Region Mapping and a Time Series
+            projects = dao.get_all_projects()
+            for project in projects:
+                datatypes = dao.get_datatypes_in_project(project.id)
```

Comment:
Waw. These queries are extremely costly. 
Why not to keep (maybe fix) the original intended query ?

Reply:
Because that query was deleting the links from the project to be deleted and after I refactored the code by the time it would reach that line, those links are already deleted. My queries are intending to remove redundant links: links which point to the same project that the datatype is in. I will think of doing it without these 3 fors...